Native module install errors with latest nightly release

<!--

Have you read Atom's Code of Conduct? By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/atom/.github/blob/master/CODE_OF_CONDUCT.md

Do you want to ask a question? Are you looking for support? The Atom message board is the best place for getting support: https://discuss.atom.io

-->

### Prerequisites

na

### Description

<!-- Description of the issue -->
When building a native module, it will throw the following error
```
gyp: Call to 'node -e "require('nan')"' returned exit status 134 while in binding.gyp. while trying to load binding.gyp
```
This call appears in the `binding.gyp` of most native modules, but not the test module bundled in apm, so the build tools check does not pick it up.

### Steps to Reproduce

1. Run `apm-nightly install termination`
2. Observe error similar to following
<details>
  <summary>Error message</summary>

```
22:35 $ apm-nightly install termination
Installing termination to /home/benjamin/.atom/packages ✗

> node-pty-prebuilt@0.7.6 install /tmp/apm-install-dir-119624-10760-7pdou5.9pgq/node_modules/termination/node_modules/node-pty-prebuilt
> prebuild-install || node scripts/install.js


Aborted (core dumped)
Aborted (core dumped)
gyp: Call to 'node -e "require('nan')"' returned exit status 134 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/share/atom-nightly/resources/app/apm/node_modules/npm/node_modules/node-gyp/lib/configure.js:345:16)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:237:12)
gyp ERR! System Linux 5.0.0-21-generic
gyp ERR! command "/usr/share/atom-nightly/resources/app/apm/bin/node" "/usr/share/atom-nightly/resources/app/apm/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--build_v8_with_gn=false"
gyp ERR! cwd /tmp/apm-install-dir-119624-10760-7pdou5.9pgq/node_modules/termination/node_modules/node-pty-prebuilt
gyp ERR! node -v v10.2.1
gyp ERR! node-gyp -v v3.7.0
gyp ERR! not ok 
Aborted (core dumped)
Aborted (core dumped)
npm WARN enoent ENOENT: no such file or directory, open '/tmp/apm-install-dir-119624-10760-7pdou5.9pgq/package.json'
npm WARN apm-install-dir-119624-10760-7pdou5.9pgq No description
npm WARN apm-install-dir-119624-10760-7pdou5.9pgq No repository field.
npm WARN apm-install-dir-119624-10760-7pdou5.9pgq No README data
npm WARN apm-install-dir-119624-10760-7pdou5.9pgq No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 134
npm ERR! node-pty-prebuilt@0.7.6 install: `prebuild-install || node scripts/install.js`
npm ERR! Exit status 134
npm ERR! 
npm ERR! Failed at the node-pty-prebuilt@0.7.6 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/benjamin/.atom/.apm/_logs/2019-07-24T12_36_29_137Z-debug.log
```

</details>

**Expected behavior:**

Build without issue

**Actual behavior:**

Error

**Reproduces how often:**

100%

### Versions

```
22:40 $ apm-nightly -v
apm  2.4.2
npm  6.2.0
node 10.2.1 x64
atom 1.41.0-nightly1
python 2.7.16
git 2.20.1
```

### Additional Information

This issue does not occur with apx. AFAIK, the most significant difference is for apx I pass the build config via [environment](https://github.com/Aerijo/apx/blob/3a9fe2f5036b6a93019bd083f31fe0dca5611b53/src/install.ts#L34) [variables](https://github.com/Aerijo/apx/blob/3a9fe2f5036b6a93019bd083f31fe0dca5611b53/src/context.ts#L219), and no flags.

--- a/src/command.coffee
+++ b/src/command.coffee
@@ -106,6 +106,11 @@ class Command
     @updateWindowsEnv(env) if config.isWin32()
     @addNodeBinToEnv(env)
     @addProxyToEnv(env)
+    env.npm_config_runtime = "electron"
+    env.npm_config_target = @electronVersion
+    env.npm_config_disturl = config.getElectronUrl()
+    env.npm_config_arch = config.getElectronArch()
+    env.npm_config_target_arch = config.getElectronArch() # for node-pre-gyp
 
   getVisualStudioFlags: ->
     if vsVersion = config.getInstalledVisualStudioFlag()

